code/game/machinery/_machinery.dm

```diff
@@ -955,11 +955,36 @@
 	return TRUE
 
 /obj/machinery/proc/display_parts(mob/user)
-	. = list()
-	. += span_notice("It contains the following parts:")
-	for(var/obj/item/C in component_parts)
-		. += span_notice("[icon2html(C, user)] \A [C].")
-	. = jointext(., "")
+	var/list/part_count = list()
+	for(var/obj/item/component_part in component_parts)
+		if(!part_count[component_part.name])
```

Comment:
Early-return

Reply:
part_count is an temporary map where the key is the part name and the value is its amount in the machine e.g part_count["matter bin"]=2 means there are 2 matter bins. 

The logic is "If this is my first time encountering this part(I.e !part_count) them set its amount to 1 else if have encountered this part before then increment that count

Again same as my previous comment if I return here then It would not count remaining parts in the for loop and the code would break.  

The for loop must complete for all parts to be counted and can't be broken anywhere

Reply:
Early-continue is what I meant, which is possible